### PR TITLE
BUGFIX: Allow disabling of auto-created Image Variants

### PR DESCRIPTION
**Pull request by bwaidelich**
Fixes support for the setting `autoCreateImageVariantPresets` that was documented for a long time but never actually evaluated.

This change set:

* Adjusts `AssetService::assetCreated()` signal to only trigger `AssetVariantGenerator::createVariants()` if the `autoCreateImageVariantPresets` flag is set
* Sets the default value of the flag to `true` for greater backwards compatibility
* Adjusts `AssetVariantGenerator::createVariant()` to only create a variant if it does not exist already – previously multiple variants with the same identifiers could be created for a single asset leading to undeterministic behavior
* Adds a button "Create missing Variants" to the `Variants` tab of the Media Module allowing editors to manually trigger creation of (missing) variants.

Fixes: #4300

### CODE REVIEW

File: Neos.Media/Classes/Domain/Service/AssetVariantGenerator.php [hunk 1]
```diff
@@ -127,6 +127,12 @@ public function createVariant(AssetInterface $asset, string $presetIdentifier, s
         $createdVariant = null;
         $preset = $this->getVariantPresets()[$presetIdentifier] ?? null;
         if ($preset instanceof VariantPreset && $preset->matchesMediaType($asset->getMediaType())) {
+
+            $existingVariant = $asset->getVariant($presetIdentifier, $variantIdentifier);
+            if ($existingVariant !== null) {
+                return $existingVariant;
+            }
```
Comment (ahaeslich) [inline Neos.Media/Classes/Domain/Service/AssetVariantGenerator.php:133]: Now there is actually a difference between `createVariant` and `recreateVariant`. Why didn't I notice this before 🤔? 

Now calling `./flow media:renderVariants` without `--recreate` actually only creates missing variants. 💖

File: Neos.Media.Browser/Classes/Controller/AssetController.php [hunk 2]
```diff
@@ -468,6 +475,32 @@ public function variantsAction(string $assetSourceIdentifier, string $assetProxy
         }
     }
 
+    /**
+     * (Re-)create all variants for the given image
```
Comment (ahaeslich) [inline Neos.Media.Browser/Classes/Controller/AssetController.php]: Did you intend to add an option to force recreation of asset variants with this action or is this just a missleading comment?

**Review by ahaeslich — APPROVED** [review]
Works like a charm. Just tested it in a project 💖

File: Neos.Media.Browser/Resources/Private/Translations/en/Main.xlf [hunk 3]
```diff
@@ -427,6 +427,9 @@
 			<trans-unit id="missingDocumentNode" xml:space="preserve">
 				<source>No document node found for this node</source>
 			</trans-unit>
+			<trans-unit id="createMissingVariants" xml:space="preserve">
+				<source>Create missing Variants</source>
```
Comment (lorenzulrich) [inline Neos.Media.Browser/Resources/Private/Translations/en/Main.xlf]: variants should be in lowercase. Neos doesn't use title case in language labels as far as I know.

**Review by lorenzulrich — APPROVED** [review]

**Comment by lorenzulrich** [comment]
@bwaidelich Thanks a lot, that's great to have.

**Comment by mhsdesign** [comment]
Hi ;) i just found this pr - were you able to test this @lorenzulrich and is this ready to be merged @bwaidelich - maybe we can include it in the upcoming bugfix release (today or tomorrow).

**Comment by bwaidelich** [comment]
Yes, this is good to go IITC

File: Neos.Media.Browser/Resources/Private/Translations/en/Main.xlf (hunk 3 above)
Comment (bwaidelich) [inline Neos.Media.Browser/Resources/Private/Translations/en/Main.xlf]:
```suggestion
				<source>Create missing variants</source>
```

File: Neos.Media.Browser/Classes/Controller/AssetController.php (hunk 2 above)
Comment (bwaidelich) [inline Neos.Media.Browser/Classes/Controller/AssetController.php]: You're right, that was misleading. Is this better:
```suggestion
     * Create missing variants for the given image
```

**Review by bwaidelich — COMMENTED** [review]
Tweak according to reviews

**Review by kdambekalns — APPROVED** [review]
LGTM

**Pull request merged**